src/network/mod.rs

```diff
@@ -328,15 +356,19 @@ pub fn bootstrap(config: &Configuration, blockchain: BlockchainR) {
 /// The calling thread is blocked until the block is retrieved.
 /// This function is called during blockchain initialization
 /// to retrieve the genesis block.
-pub fn fetch_block(config: &Configuration, hash: &HeaderHash) -> Result<Block, FetchBlockError> {
+pub fn fetch_block(
+    config: &Configuration,
+    hash: &HeaderHash,
+    logger: &Logger,
+) -> Result<Block, FetchBlockError> {
     if config.protocol != Protocol::Grpc {
         unimplemented!()
     }
     match first_trusted_peer_address(config) {
         None => Err(FetchBlockError::NoTrustedPeers),
         Some(address) => {
             let peer = Peer::new(address, Protocol::Grpc);
-            grpc::fetch_block(peer, hash)
+            grpc::fetch_block(peer, hash, logger)
```

Comment:
This should be a cloned and augmented instance, similarly to the bootstrap task above.

Reply:
Will fix it like bootstrap, in `main::initialize_node`